lib/watirspec/implementation.rb

```diff
@@ -8,6 +8,11 @@ def initialize
       @guard_proc = nil
     end
 
+    def initialize_copy(orig)
+      super
+      @browser_args = Marshal.load( Marshal.dump(@browser_args))
```

Comment:
This was a pain to figure out. A clone or dup is a different object referencing the same values inside the browser_args array. I was really struggling especially when I was trying to do this in the specs themselves. Now that it is implemented inside the class with `#initialize_copy`, I see that I can update it to be: `@browser_args = browser_args.map(&:dup)`. Thanks for making me check again, this is much better than Marshal.